File: 6 Kyu/dig_pow.py
# 03/02/19

'''
  6 Kyu
  Problem provided by https://www.codewars.com/kata/playing-with-digits
'''

def dig_pow(n, p):
	n_digits_array = [int(x) for x in str(n)]
	n_sum = 0

	for digit in n_digits_array:
		n_sum += pow(digit, p)
		p += 1

	if n_sum % n == 0:
		return n_sum / n

	return -1

File: 6 Kyu/digPow.js
// 03/02/19

/*
  6 Kyu
  Problem provided by https://www.codewars.com/kata/playing-with-digits
*/

function digPow(n, p) {
  // Create an array of the digits of n.
  let nDigitsArray = Array.from(n.toString()).map(Number);

  let nSum = 0;
  for(let i = 0; i < nDigitsArray.length; i++) {
    nSum+=(Math.pow(nDigitsArray[i], p));
    p++;
  }

  if(Number.isInteger(nSum / n))
    return nSum / n;

  return -1;
}
